src/Utilities/Idm/IdmLogger.f90

```diff
@@ -324,4 +331,237 @@ subroutine idm_log_var_str(p_mem, varname, mempath, iout)
+    integer(I4B), intent(in) :: iout
+    ! -- dummy
+    integer(I4B) :: i, j, inunit
+    !
+    do i = 1, size(p_mem, dim=2)
+      inunit = create_export_file(varname, mempath, i, iout)
+      write (inunit, '(*(i0, " "))') (p_mem(j, i), j=1, size(p_mem, dim=1))
+      close (inunit)
+    end do
+  end subroutine idm_export_int2d
+
+  !> @brief Create export file int3d
+  !<
+  subroutine idm_export_int3d(p_mem, varname, mempath, iout)
+    integer(I4B), dimension(:, :, :), contiguous, intent(in) :: p_mem !< 2d dbl array
+    character(len=*), intent(in) :: varname !< variable name
+    character(len=*), intent(in) :: mempath !< variable memory path
+    integer(I4B), intent(in) :: iout
+    ! -- dummy
+    integer(I4B) :: i, j, k, inunit
+    !
+    do k = 1, size(p_mem, dim=3)
+      inunit = create_export_file(varname, mempath, k, iout)
+      do i = 1, size(p_mem, dim=2)
+        write (inunit, '(*(i0, " "))') (p_mem(j, i, k), j=1, size(p_mem, dim=1))
+      end do
+      close (inunit)
+    end do
+  end subroutine idm_export_int3d
+
+  !> @brief Create export file dbl1d
+  !!
+  !! export layered dbl1d parameters with NODES shape
+  !!
+  !<
+  subroutine idm_export_dbl1d(p_mem, varname, mempath, iout)
+    use SimVariablesModule, only: idm_context
+    use ConstantsModule, only: LENMEMPATH, LENCOMPONENTNAME
+    use MemoryManagerModule, only: mem_setptr
+    use MemoryHelperModule, only: create_mem_path, split_mem_path
+    real(DP), dimension(:), contiguous, intent(in) :: p_mem !< 1d dbl array
+    character(len=*), intent(in) :: varname !< variable name
+    character(len=*), intent(in) :: mempath !< variable memory path
+    integer(I4B), intent(in) :: iout
+    ! -- dummy
+    integer(I4B), dimension(:), pointer, contiguous :: model_shape
+    real(DP), dimension(:, :, :), pointer, contiguous :: dbl3d
+    real(DP), dimension(:, :), pointer, contiguous :: dbl2d
+    integer(I4B), dimension(3) :: dis_shape
+    integer(I4B), dimension(2) :: disv_shape
+    character(LENMEMPATH) :: input_mempath
+    character(LENCOMPONENTNAME) :: comp, subcomp
+    integer(I4B) :: i, j, k, inunit
+    !
+    call split_mem_path(mempath, comp, subcomp)
+    input_mempath = create_mem_path(component=comp, context=idm_context)
+    call mem_setptr(model_shape, 'MODEL_SHAPE', input_mempath)
+    !
+    if (size(model_shape) == 3) then
+      dis_shape(1) = model_shape(3)
+      dis_shape(2) = model_shape(2)
+      dis_shape(3) = model_shape(1)
+      allocate (dbl3d(dis_shape(1), dis_shape(2), dis_shape(3)))
+      dbl3d = reshape(p_mem, dis_shape)
+      do k = 1, dis_shape(3)
+        inunit = create_export_file(varname, mempath, k, iout)
+        do i = 1, model_shape(2)
+          write (inunit, '(*(G0.10, " "))') (dbl3d(j, i, k), j=1, &
+                                             dis_shape(1))
+        end do
+        close (inunit)
+      end do
+      deallocate (dbl3d)
+    else if (size(model_shape) == 2) then
```

Comment:
As the code is currently written, these routines won't be called for any Dis2D parameters because none support a layered read.  Is this sufficient for now?  This will need to change with the aim of broader array export support.